prime-angular-frontend/src/app/modules/site-registration/shared/guards/organization.guard.ts

```diff
@@ -37,22 +38,24 @@ export class OrganizationGuard extends BaseGuard {
         // Having no signing authority or organizations results in the same
         // redirection logic for the user, and therefore not handled individually
         exhaustMap((user: BcscUser) =>
-          this.organizationResource.getSigningAuthorityOrganizationsByUserId(user.userId)
+          forkJoin([
+            this.organizationResource.getSigningAuthorityOrganizationsByUserId(user.userId),
+            this.organizationResource.getSigningAuthorityByUserId(user.userId),
+            this.organizationResource.getOrganizationClaim({ userId: user.userId })
+          ])
         ),
-        map((organizations: Organization[] | null) =>
-          (organizations?.length)
-            ? organizations.shift()
-            : null
+        map((result: [Organization[] | null, Party, boolean]) =>
+          [result[0]?.length ? result[0].shift() : null, result[1], result[2]]
         ),
-        map((organization: Organization | null) => {
+        map((result: [Organization | null, Party, boolean]) => {
```

Comment:
Use array destructuring instead of array indexing:

```
map(({ organization, party, claimed }: [Organization | null, Party, boolean]) => ...
```